core/state/statedb.go

```diff
@@ -853,7 +850,6 @@ func (s *StateDB) Finalise(deleteEmptyObjects bool) {
 		} else {
 			obj.finalise(true) // Prefetch slots in the background
 		}
-		obj.created = false
```

Comment:
Doesn't this change the logic? Previously, the `created` field would always be unset. With the new way, it only gets reset if the account did not self destruct.